packages/backend-common/src/reading/AwsS3UrlReader.ts

```diff
@@ -215,44 +277,22 @@ export class AwsS3UrlReader implements UrlReader {
           Key: path,
         };
       }
-
-      const request = this.deps.s3.getObject(params);
-      options?.signal?.addEventListener('abort', () => request.abort());
-
-      // Since we're consuming the read stream we need to consume headers and errors via events.
-      const etagPromise = new Promise<string | undefined>((resolve, reject) => {
-        request.on('httpHeaders', (status, headers) => {
-          if (status < 400) {
-            if (status === 200) {
-              resolve(headers.etag);
-            } else if (status !== 304 /* not modified */) {
-              reject(
-                new Error(
-                  `S3 readUrl request received unexpected status '${status}' in response`,
-                ),
-              );
-            }
-          }
-        });
-        request.on('error', error => reject(error));
-        request.on('complete', () =>
-          reject(
-            new Error('S3 readUrl request completed without receiving headers'),
-          ),
-        );
+      options?.signal?.addEventListener('abort', () => abortController.abort());
```

Comment:
I see you make extra AbortController instances that do nothing but mirror the "incoming" signal. Is the signal to the S3 API mandatory, is that why?

Reply:
The previous implementation had an abort listener, and the AWS SDK has a Controller to support the action, so it seemed like adding the `AbortController` would keep the same existing behavior. However, the signal isn't mandatory; we can remove it safely.